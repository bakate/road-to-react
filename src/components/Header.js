import React from 'react'

const Header = ({ text }) => {
  return (
    <button type="button" name="title" data-testid="h1bb" className="yo">
      I'm {text}
    </button>
  )
}

export default Header
