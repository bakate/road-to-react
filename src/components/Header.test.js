import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import React from 'react'
import Header from './Header'

it('renders', () => {
  const { asFragment } = render(<Header text="Bakate" />)
  expect(asFragment()).toMatchSnapshot()
})
it('renders', () => {
  const { getByTestId, getByText } = render(<Header text="Bakate" />)
  expect(getByTestId('h1bb')).toHaveTextContent("I'm Bakate")
  expect(getByText("I'm Bakate")).toHaveClass('yo')
})
