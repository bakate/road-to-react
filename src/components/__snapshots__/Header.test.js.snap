// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    class="yo"
    data-testid="h1bb"
    name="title"
    type="button"
  >
    I'm Bakate
  </button>
</DocumentFragment>
`;
